src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1311 @@
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: T1 = S._copy_()
+            sage: T1.mutate(0, inplace=True)
+            A seed with cluster variables [(x2 + 1)/x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+            sage: T2 = S.mutate(0, inplace=False)
+            sage: T1==T2
+            True
+        """
+        return LPASeed(self)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        TESTS::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: print(LPASeed({x1: 1 + x2, x2: 1 + x1}))
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+        """
+        return ("A seed with cluster variables {0}"
+                " and exchange polynomials {1}").format(self.cluster(),
+                                                        self.exchange_polys())
+
+    def __hash__(self):
+        """
+        Return a hash of ``self``.
+
+        EXAMPLES:
+
+        We check that two seeds with the same hash are equal::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: T = LPASeed(S)
+            sage: S.mutate([0,1,0])
+            A seed with cluster variables [(x1 + 1)/x2, (x1 + x2 + 1)/(x1*x2)] and exchange polynomials [x2 + 1, x1 + 1]
+            sage: T.mutate([1,0])
+            A seed with cluster variables [(x1 + x2 + 1)/(x1*x2), (x1 + 1)/x2] and exchange polynomials [x2 + 1, x1 + 1]
+            sage: hash(S) == hash(T)
+            True
+            sage: S == T
+            True
+        """
+        return hash(frozenset(self.cluster()))
+
+    def _latex_(self):
+        r"""
+        Returns a `\LaTeX` representation of this seed.
+
+        TESTS::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: latex(LPASeed({x1: 1 + x2, x2: 1 + x1}))
+            \left(\left(x_{1}, x_{2} + 1\right), \left(x_{2}, x_{1} + 1\right)\right)
+
+        """
+        cvars = self.cluster()
+        polys = self.exchange_polys()
+        cvar_poly_pairs = list()
+        for i in range(self.rank()):
+            cvar_poly_pairs.append(latex(tuple([cvars[i], polys[i]])))
+
+        return latex(tuple(cvar_poly_pairs))
+
+    def are_laurent_polys_trivial(self):
+        r"""
+        Returns whether this seed has Laurent polynomials with nontrivial denominators.
+        """
+        for poly in self.laurent_polys():
+
+            if poly.denominator() != 1:
+
+                return False
+
+        return True
+
+    def is_mutation_infinite(self, give_reason=True):
+        r"""
+        Performs some heuristic checks on the mutation class of this seed to work out if it is mutation infinite or not.
+        Completely probabilistic and may false positive; do not use in research setting.
+        Warning: This could take a long time for some seeds!
```

Comment:
```suggestion
        Performs some heuristic checks on the mutation class of this seed to
        determine out if it is mutation infinite or not.

       .. WARNING::

            This uses a probabilistic algorithm that may give a false positive.

                
        .. NOTE::

            This could take a long time for some seeds!
```